platform/ios/vendor/SMCalloutView/SMCalloutView.m

```diff
@@ -554,13 +554,18 @@ - (CAAnimation *)animationWithType:(MGLSMCalloutAnimation)type presenting:(BOOL)
         stretch.toValue = presenting ? @1.0 : @0.0;
         animation = stretch;
     }
-    
+
+    NSAssert(animation, @"There should be an animation");
```

Comment:
Instead of asserting and casting, convert the if/else-if structure above into an exhaustive switch statement over `type`.